src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/Rss20FeedFormatter.cs

```diff
@@ -302,8 +302,16 @@ private SyndicationLink ReadAlternateLink(XmlReader reader, Uri baseUri)
                 }
             }
 
-            string uri = reader.ReadElementString();
-            link.Uri = UriParser(uri, UriKind.RelativeOrAbsolute, Rss20Constants.LinkTag, Rss20Constants.Rss20Namespace);
+            string uriString = reader.ReadElementString();
+            var elemntQualifiedName = new XmlQualifiedName(Rss20Constants.LinkTag, Rss20Constants.Rss20Namespace);
+            var parseUriArgs = new ParseUriArgs()
+            {
+                UriString = uriString,
+                UriKind = UriKind.RelativeOrAbsolute,
+                ElemntQualifiedName = elemntQualifiedName
+            };
+            uriParser(parseUriArgs, out Uri uri);
+            link.Uri = uri;
```

Comment:
This doesn't handle the multiple delegates and exception handling that UriFromString provides. Consider making UriFromString static and have it accept TryParseUri as a parameter. Then make another UriFromString which isn't static which calls the static version passing in the instance of the uri parser.

Reply:
Good catch. Will fix this. Thanks.